[invalid] Test uploads with azcopy

dg0yt opened the pull request:
Complements https://github.com/microsoft/vcpkg-tool/pull/1679.

dg0yt committed:
[invalid] Test uploads with azcopy

dg0yt committed:
Test vcpkg without azcopy capability

dg0yt committed:
Revert "Test vcpkg without azcopy capability"

This reverts commit 35b7dffd0280094eb64dadeb8de52de5ba5e7a6f.

dg0yt committed:
WIP

dg0yt committed:
Use vcpkg fetch

dg0yt commented:
So, osx is not starting with pristine images...

dg0yt commented on scripts/azure-pipelines/test-modified-ports.ps1:
```diff
@@ -128,6 +128,59 @@ if ($Triplet -eq 'x64-windows-release') {
+    exit $lastLastExitCode
+}
+$azcopy
+$env:PATH += [IO.Path]::PathSeparator + ( Split-Path -Parent ( $azcopy | Select-Object -Last 1 ) )
```
`Select-Object -Last 1` is for https://github.com/microsoft/vcpkg/pull/43309/files#r1952400952.

dg0yt committed:
Fixup

dg0yt commented:
> So, osx does not use pristine images...

It really does not: It uninstalls 368 packages which it didn't install in the same CI run in https://dev.azure.com/vcpkg/public/_build/results?buildId=115637&view=logs&j=ac0731cb-f2f9-5764-2e0a-a6fa8ebb4a08&t=1f8aab00-85d2-59d6-ec59-5750d1aa4b81
CC @BillyONeal

xavier2k6 committed:
Update `azcopy` to `10.29.1` (vcpkg-tools.json)

dg0yt committed:
Merge remote-tracking branch 'origin/master' into azcopy

dg0yt committed:
Merge branch 'vcpkg_azcopy' of https://github.com/xavier2k6/vcpkg into azcopy

vicroms commented:
I missed that you had already created this PR, thanks for the heads-up! 
We're ok with this test using artificial data.

dg0yt commented:
Remature close? It is not yet used in this repo. Since users can't reopen PRs, I would need to open a new one if I want to do the next round of testing.

BillyONeal commented:
> Remature close? It is not yet used in this repo. Since users can't reopen PRs, I would need to open a new one if I want to do the next round of testing.

I understood this as 'only to demonstrate that https://github.com/microsoft/vcpkg-tool/pull/1679 is sane'

dg0yt committed:
Merge remote-tracking branch 'origin/master' into azcopy

dg0yt committed:
Use default tool

dg0yt committed:
Skip fetching azcopy

dg0yt committed:
Test

dg0yt commented:
> for now we should add `azcopy` to the image.

@BillyONeal Will you update the images? Windows patch day is next week, but there isn't only Windows.
In addition, two environment variables must be set for azcopy to work, at least on some platforms.

dg0yt committed:
WIP

vicroms committed:
Use azcopy for binary caching

vicroms committed:
Test with zlib

dg0yt commented on scripts/azure-pipelines/test-modified-ports.ps1:
```diff
@@ -128,6 +128,59 @@ if ($Triplet -eq 'x64-windows-release') {
+}
+$azcopy
+$env:PATH += [IO.Path]::PathSeparator + ( Split-Path -Parent ( $azcopy | Select-Object -Last 1 ) )
+$env:AZCOPY_LOG_LOCATION = Join-Path $ArtifactStagingDirectory 'azcopy-logs'
+$env:AZCOPY_JOB_PLAN_LOCATION = Join-Path $ArtifactStagingDirectory 'azcopy-plans'
```
@vicroms

vicroms committed:
Set azcopy log and plan location

vicroms committed:
Use test port

vicroms committed:
Trigger CI

vicroms committed:
Trigger CI

vicroms committed:
Downgrade to 10.28.0

vicroms committed:
Use azcopy-sas for all platforms

dg0yt committed:
Merge branch 'azcopy-test' of https://github.com/vicroms/vcpkg into azcopy

dg0yt committed:
Revert vcpkg-ci-boost

dg0yt committed:
WIP

dg0yt commented:
No, x64-osx fails to upload:
~~~
warning: error: azcopy failed with exit code: (1).
~~~

vicroms commented:
/azp run

vicroms committed:
Update scripts/azure-pipelines/test-modified-ports.ps1

vicroms commented on scripts/azure-pipelines/test-modified-ports.ps1:
```diff
@@ -85,7 +85,7 @@ $buildtreesRoot = Join-Path $WorkingRoot 'b'
 
-$env:AZCOPY_LOG_LOCATION = Join-Path $WorkingRoot 'azcopy-logs'
+$env:AZCOPY_LOG_LOCATION = Join-Path $ArtifactStagingDirectory 'azcopy-logs'
 $env:AZCOPY_JOB_PLAN_LOCATION = Join-Path $WorkingRoot 'azcopy-plans'
```
I was unsure if I wanted to publish AzCopy logs and plans from the build since I was cautious about leaking SAS tokens. After some local testing, I think it is safe to publish the logs but not the plans, but some more work is required to publish them still.

vicroms committed:
Publish azcopy logs

vicroms committed:
Merge branch 'azcopy' of https://github.com/dg0yt/vcpkg into azcopy

vicroms committed:
Publish azcopy logs

dg0yt committed:
Update scripts/azure-pipelines/azure-pipelines.yml

dg0yt committed:
Restore azcopy 10.29.1

dg0yt committed:
Update azcopy to 10.30.0

dg0yt committed:
Trace

vicroms commented:
Now that we have logs, I can ask the AzCopy team for their input. I did a quick search in StackOverflow and some answers suggest limiting the number of concurrent connections.

dg0yt commented:
> I did a quick search in StackOverflow and some answers suggest limiting the number of concurrent connections.

In https://github.com/microsoft/vcpkg-tool/pull/1679#issuecomment-2875117020, I asked/warned:

> Would vcpkg run multiple instances of azcopy at the same time? This doesn't seem to be the pattern preferred by azcopy.

This was less likely to occur with x-azblob only offloading large artifacts, but x-azcopy uses aczopy for all artifacts.

dg0yt committed:
Merge branch 'microsoft:master' into azcopy

dg0yt committed:
Limit azcopy buffer and concurrency

https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-optimize

dg0yt commented on scripts/azure-pipelines/azure-pipelines.yml:
```diff
@@ -5,11 +5,11 @@ parameters:
     displayName: 'Custom SHA of vcpkg-tool to use rather than bootstrap'
     type: string
-    default:  'use default'
+    default:  'main'
```
```suggestion
    default:  'use default'
```

dg0yt committed:
Apply suggestions from code review

dg0yt committed:
Merge branch 'microsoft:master' into azcopy

dg0yt commented:
azcopy is live.
